cmd/simulator/load/loader.go

```diff
@@ -115,12 +119,15 @@ func ExecuteLoader(ctx context.Context, config config.Config) error {
 		agents = append(agents, txs.NewIssueNAgent[*types.Transaction](txSequences[i], NewSingleAddressTxWorker(ctx, clients[i], senders[i]), config.BatchSize))
 	}
 
+	reg := prometheus.NewRegistry()
+	m := metrics.NewMetrics(reg)
+
 	log.Info("Starting tx agents...")
 	eg := errgroup.Group{}
 	for _, agent := range agents {
 		agent := agent
 		eg.Go(func() error {
-			return agent.Execute(ctx)
+			return agent.Execute(ctx, m)
```

Comment:
can we add metrics to the constructor of a new agent instead of a parameter to `Execute`

Reply:
yup